### PR TITLE
Refactor FXIOS-2040 [v104] Removed SnapKit from TabLocationView, TabToolbar, and TabToolbarHelper

### PR DESCRIPTION
**Pull request by KrisRJack**
Related to https://github.com/mozilla-mobile/firefox-ios/issues/8592

This PR includes replacement for SnapKit in TabLocationView, TabToolbar, and TabToolbarHelper. 

Additionally, this PR includes two helper functions in UIView extensions:
- `edges(equalTo: UIView, padding: CGFloat)` - Makes the edge constraints of a view equaled to the edge constrains of another view.
-  `center(equalTo: UIView)` - Makes x and y anchors of a view equaled to that of another view.

### CODE REVIEW

File: Client/Frontend/Toolbar+URLBar/TabLocationView.swift [hunk 1]
```diff
@@ -142,35 +134,26 @@ class TabLocationView: UIView {
...
 
-        contentView.snp.makeConstraints { make in
-            make.edges.equalTo(self)
-        }
-
-        trackingProtectionButton.snp.makeConstraints { make in
-            make.width.equalTo(TabLocationViewUX.TPIconSize)
-            make.height.equalTo(TabLocationViewUX.ButtonSize)
-        }
+        contentView.edges(equalTo: self)
```
Comment (KrisRJack) [inline Client/Frontend/Toolbar+URLBar/TabLocationView.swift:147]: This function was created [here](https://github.com/mozilla-mobile/firefox-ios/blob/fc4eb7dee8077f11a36ee76f180913af0b65c5ab/Client/Extensions/UIViewExtensions.swift#L74)

File: Client/Extensions/UIViewExtensions.swift [hunk 2]
```diff
@@ -67,6 +67,28 @@ extension UIView {
...
+            topAnchor.constraint(equalTo: view.topAnchor, constant: constant),
+            bottomAnchor.constraint(equalTo: view.bottomAnchor, constant: constant),
+            leadingAnchor.constraint(equalTo: view.leadingAnchor, constant: constant),
+            trailingAnchor.constraint(equalTo: view.trailingAnchor, constant: constant)
+        ])
+    }
+    
+    /**
+     * Makes x and y anchors of a view equaled to that of another view.
+     */
```
Comment (electricRGB) [inline Client/Extensions/UIViewExtensions.swift:85]: Same comment as above re docs. :)

File: Client/Frontend/Toolbar+URLBar/TabLocationView.swift [hunk 3]
```diff
@@ -142,35 +134,26 @@ class TabLocationView: UIView {
...
-            make.height.equalTo(TabLocationViewUX.ButtonSize)
-        }
-
-        reloadButton.snp.makeConstraints { make in
-            make.width.equalTo(TabLocationViewUX.ReaderModeButtonWidth)
-            make.height.equalTo(TabLocationViewUX.ButtonSize)
-        }
+        NSLayoutConstraint.activate([
+            trackingProtectionButton.widthAnchor.constraint(equalToConstant: TabLocationViewUX.TPIconSize),
+            trackingProtectionButton.heightAnchor.constraint(equalToConstant: TabLocationViewUX.ButtonSize),
```
Comment (electricRGB) [inline Client/Frontend/Toolbar+URLBar/TabLocationView.swift:151]:
```suggestion
            trackingProtectionButton.heightAnchor.constraint(equalToConstant: TabLocationViewUX.ButtonSize),

```
nit: vertical space is cheap and it helps improve legibility by separating constraints for different objects.

**Review by electricRGB — CHANGES_REQUESTED** [review]
Heya. Thanks for your continuing contributions! Just a few comments to slightly improve some of the good stuff here, before I get to running this locally & on BR so we can eventually merge.

File: Client/Extensions/UIViewExtensions.swift [hunk 4]
```diff
@@ -67,6 +67,28 @@ extension UIView {
         maskLayer.path = maskPath.cgPath
         layer.mask = maskLayer
     }
+    
+    /**
+     * Makes the edge constraints (`topAnchor`, `bottomAnchor`, `leadingAnchor`, `trailingAnchor`) of a view equaled to the edge constrains of another view.
+     */
```
[the remark this answers is not in the document]
Comment (KrisRJack) [reply Client/Extensions/UIViewExtensions.swift]: Neat! Will use `Option + Command + /` more often. Did not know that shortcut existed 😅

**Review by electricRGB — APPROVED** [review]

**Comment by electricRGB** [comment]
BR is being silly. Closing & reopening to re-run.

**Comment by electricRGB** [comment]
BR is green

Screen Shot 2022-07-11 at 2 20 26 PM

**Comment by KrisRJack** [comment]
@electricRGB Done! Thanks for your review!

**Comment by electricRGB** [comment]
The test is passing locally.
Screen Shot 2022-07-14 at 10 40 37 AM

**Pull request merged**